frame/executive/src/lib.rs

```diff
@@ -354,7 +356,12 @@ where
 		});
 
 		// post-extrinsics book-keeping
+
 		<frame_system::Module<System>>::note_finished_extrinsics();
+
+		<frame_system::Module<System> as OnIdle<System::BlockNumber>>::on_idle(block_number);
```

Comment:
here you should take currently used weight from `frame_system::BlockWeight` storage.
And looks at was is remaining by substracting it from `frame_system::Config::BlockWeights::get().max_block`.

Or we can be more precise and give what is remaining for normal space and operational space.